compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -765,13 +754,36 @@ trait ParallelTesting extends RunnerOrchestration { self =>
           }
         }
 
+        def fail(msg: String): Unit = {
+          echo(msg)
+          failTestSource(testSource)
+        }
+
+        def reporterOutputLines(reporter: TestReporter): List[String] = {
+          reporter.allErrors.flatMap { error =>
+            (error.pos.span.toString + " in " + error.pos.source.file.name) :: error.getMessage().split("\n").toList
+          }
+        }
+        def checkFileTest(sourceName: String, checkFile: JFile, actual: List[String]) = {
+          if (checkFile.exists) {
+            val expexted = Source.fromFile(checkFile, "UTF-8").getLines().toList
+            diffMessage(sourceName, actual, expexted).foreach(fail)
+          }
+        }
+
         val (compilerCrashed, expectedErrors, actualErrors, hasMissingAnnotations, errorMap) = testSource match {
           case testSource @ JointCompilationSource(_, files, flags, outDir, fromTasty, decompilation) =>
             val sourceFiles = testSource.sourceFiles
             val (errorMap, expectedErrors) = getErrorMapAndExpectedCount(sourceFiles)
             val reporter = compile(sourceFiles, flags, true, outDir)
             val actualErrors = reporter.errorCount
-
+            files.foreach { file =>
```

Comment:
`sourceFiles`? => no special case for directory

Reply:
That is the way it works for `pos` and `run` tests.